### PR TITLE
GRVDEV-160 Update S3-CDN Module to Include Edge Lambda to Apply Security Headers

### PR DESCRIPTION
**Pull request by TysonBeck**

### CODE REVIEW

File: aws/s3-cdn/s3-cdn.tf [hunk 1]
```diff
@@ -302,6 +302,34 @@ variable "kms_master_key_arn" {
...
+  description = "Enables creation and attachment of origin-responce edge lambda to add security headers"
+}
+
+variable nodejs_security_header_code {
+  type        = string
+  default     = <<EOF
+'use strict';
+exports.handler = (event, context, callback) => {
+    
+    //Get contents of response
```
Comment (jeremyodle) [inline aws/s3-cdn/s3-cdn.tf:317]: - [x]  Format node.js inline code

File: aws/s3-cdn/s3-cdn.tf [hunk 2]
```diff
@@ -512,6 +549,86 @@ module "dns" {
...
+
+  source_code_hash = data.archive_file.hsts.output_base64sha256
+
+  runtime     = "nodejs12.x"
+  timeout     = 1
+  memory_size = 128
+
+  tags = local.tags
+
+  tracing_config {
```
Comment (jeremyodle) [inline aws/s3-cdn/s3-cdn.tf:635]: - [x]  Determine if X-Ray tracing/logging should be enabled here or in CloudFront so tools like Datadog/AWS can get tracing/logging info

**Review by jeremyodle — CHANGES_REQUESTED** [review]

File: aws/s3-cdn/s3-cdn.tf (hunk 1 above)
Comment (TysonBeck) [reply aws/s3-cdn/s3-cdn.tf:317]: Code has been formatted as requested

File: aws/s3-cdn/s3-cdn.tf (hunk 2 above)
Comment (TysonBeck) [reply aws/s3-cdn/s3-cdn.tf:635]: Verified logs not provided when set to Active as lambda is deployed to CDN Edge. Set back to PassThrough

**Review by jeremyodle — APPROVED** [review]

**Pull request merged**